src/transformers/models/m2m_100/modeling_m2m_100.py

```diff
@@ -1491,3 +1492,284 @@ def _reorder_cache(past_key_values, beam_idx):
+            Labels for computing the masked language modeling loss. Indices should either be in `[0, ...,
+            config.vocab_size]` or -100 (see `input_ids` docstring). Tokens with indices set to `-100` are ignored
+            (masked), the loss is only computed for the tokens with labels in `[0, ..., config.vocab_size]`.
+
+        Returns:
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if labels is not None:
+            if decoder_input_ids is None:
+                decoder_input_ids = shift_tokens_right(
+                    labels, self.config.pad_token_id, self.config.decoder_start_token_id
+                )
+
+        if encoder_outputs is None:
+            raise ValueError("M2M100DecoderModel expects the `encoder_outputs` to be always present.")
+        embeddings = encoder_outputs.last_hidden_state
+        batch_size = embeddings.shape[0]
+
+        if return_dict and not isinstance(encoder_outputs, BaseModelOutput):
+            encoder_outputs = BaseModelOutput(
+                last_hidden_state=encoder_outputs[0],
+                hidden_states=encoder_outputs[1] if len(encoder_outputs) > 1 else None,
+                attentions=encoder_outputs[2] if len(encoder_outputs) > 2 else None,
+            )
+
+        # decoder outputs consists of (dec_features, past_key_value, dec_hidden, dec_attn)
+        decoder_outputs = self.decoder(
+            input_ids=decoder_input_ids,
+            attention_mask=decoder_attention_mask,
+            encoder_hidden_states=encoder_outputs[0],
+            encoder_attention_mask=attention_mask,
+            head_mask=decoder_head_mask,
+            cross_attn_head_mask=cross_attn_head_mask,
+            past_key_values=past_key_values,
+            inputs_embeds=decoder_inputs_embeds,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        lm_logits = self.lm_head(decoder_outputs[0])
+
+        masked_lm_loss = None
+        if labels is not None:
+            # move labels to the correct device to enable PP
+            labels = labels.to(lm_logits.device)
+            loss_fct = CrossEntropyLoss()
+            masked_lm_loss = loss_fct(lm_logits.view(-1, self.config.vocab_size), labels.view(-1))
+
+        if (output_attentions or self.config.output_attentions) and encoder_outputs.attentions is None:
+            # just for the sake of compatibility, adding fake encoder attentions
+            encoder_outputs.attentions = [
+                torch.ones(
+                    (batch_size, self.config.encoder_attention_heads, 1, 1),
+                    device=embeddings.device,
+                    dtype=embeddings.dtype,
+                )
+                for layer in range(self.config.encoder_layers)
+            ]
```

Comment:
Let's avoid this, we should just disable the `output_attentions` in the inputs!

Reply:
Well, disabling `output_attentions` may not be the perfect solution, because the self-attention in the decoder is still non-trivial, and the user may want to inspect it. 
But I could just remove this code and disable the corresponding unit test.